api/types/provisioning.go

```diff
@@ -69,6 +69,7 @@ func ValidateJoinMethod(method JoinMethod) error {
 // ProvisionToken is a provisioning token
 type ProvisionToken interface {
 	Resource
```

Comment:
`ResourceWithOrigin` includes `Resource`, so I imagine that you can remove the `Resource` line here.